common/src/main/java/hardcorequesting/common/client/interfaces/graphic/task/CheckBoxTaskGraphic.java

```diff
@@ -0,0 +1,37 @@
+package hardcorequesting.common.client.interfaces.graphic.task;
+
+import hardcorequesting.common.client.ClientChange;
+import hardcorequesting.common.client.interfaces.GuiQuestBook;
+import hardcorequesting.common.client.interfaces.widget.AbstractCheckBox;
+import hardcorequesting.common.network.NetworkManager;
+import hardcorequesting.common.quests.task.CheckBoxTask;
+import net.minecraft.network.chat.TextComponent;
+
+import java.util.UUID;
+
+public class CheckBoxTaskGraphic extends TaskGraphic {
+    private boolean completed;
+
+    public CheckBoxTaskGraphic(CheckBoxTask task, UUID playerId, GuiQuestBook gui) {
+        super(playerId, gui, task);
+
+        completed = task.isCompleted(playerId);
+
+        // TODO custom checkbox label?
+        addClickable(new AbstractCheckBox(gui, TextComponent.EMPTY, START_X, START_Y) {
+            @Override
+            public boolean getValue() {
+                return completed;
+            }
+
+            @Override
+            public void setValue(boolean val) {
+                if (val && !completed) {
+                    task.completeTask(playerId);
```

Comment:
I am under the impression that this call should only happen on the logical server-side.

Reply:
I'll rewrite this later to not be horrible, should probably just send a "checkbox toggled" packet to the server.

Reply:
Adding a packet for it should be good. You could either use `UPDATE_TASK` for it, or create a new packet specifically for the checkbox (which I believe would work close to the same as the `UPDATE_TASK` packet).